odilia/src/tower/handlers.rs

```diff
@@ -115,24 +115,12 @@ impl Handlers {
 		<T as TryFromState<Arc<ScreenReaderState>, C>>::Future: Send,
 		<T as TryFromState<Arc<ScreenReaderState>, C>>::Error: Send,
 	{
-		let try_cmd_layer: TryIntoLayer<C, Command> = TryIntoLayer::new();
-		let params_layer: AsyncTryIntoLayer<T, (Arc<ScreenReaderState>, C)> =
-			AsyncTryIntoLayer::new();
-		let state = Arc::clone(&self.state);
-		let state_layer: StateLayer<ScreenReaderState> = StateLayer::new(state);
-		// Service<T> -> Result<R, Infallible> -> unwrap -> R
-		// this is safe because we wrap the service in a Reuslt<R, Infallible> so that we can preserve
-		// any return type we want, including ones with no errors
-		// R -> Result<(), Error>
-		let hand_service = handler
+		let try_cmd_service = handler
 			.into_service::<R>()
-			.map_result(|r| r.expect("This should never fail").into());
-		// Service(<Arc<S>, C>) -> T
-		let params_service = params_layer.layer(hand_service);
-		// Service<C> -> (Arc<S>, C)
-		let state_service = state_layer.layer(params_service);
-		// Service<Command> -> C
-		let try_cmd_service = try_cmd_layer.layer(state_service);
+			.unwrap_map(|r| r.into())
+			.request_async_try_from()
+			.with_state(Arc::clone(&self.state))
+			.request_try_from();
 		let dn = C::CTYPE;
```

Comment:
what is this variable precisely? can you change its name to be more reflective on what it's for?

Reply:
Done in latest commits via `Chooser` and `ChooserStatic` traits.